examples/tv-casting-app/linux/main.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <controller/CHIPCommissionableNodeController.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <setup_payload/SetupPayload.h>
+#include <support/CHIPMem.h>
+
+using namespace chip;
+
+#define TV_DEVICE_TYPE 35
+
+CHIP_ERROR RunTimedEventLoop(uint32_t aDelayMilliseconds)
+{
+    System::Timer * timer = nullptr;
+    if (DeviceLayer::SystemLayer.NewTimer(timer) == CHIP_NO_ERROR)
+    {
+        timer->Start(
+            aDelayMilliseconds,
+            [](System::Layer *, void *, CHIP_ERROR err) {
+                DeviceLayer::PlatformMgr().StopEventLoopTask();
+                DeviceLayer::PlatformMgr().Shutdown();
```

Comment:
As discussed yesterday, we found a bug in src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp. The documentation is fine I think, I just got led astray by that bug and ended up using these Shutdowns because I could not get it to work any other way (I started thinking that I was misinterpreting the documentation).

I have updated the code now to not do these Shutdowns and will post a separate PR for the fix in the GenericPlatformManagerImpl_POSIX.cpp

Reply:
Just following up here. Here is the PR for the fix in the GenericPlatformManagerImpl_POSIX : https://github.com/project-chip/connectedhomeip/pull/9094